### DISCUSSION TITLE
Instantiate the Model in a loop, only the first time it succeeds

### DISCUSSION THREAD

**Discussion by yohowie** [phalcon/cphalcon#15690] [Q&A]
// Controller code 

```
$order = new Order();
$order->setTableName('order_5');
$orderInfo = $order::find('id = 1');
var_dump($orderInfo);

$order = new Order();
$order->setTableName('order_6');
$orderInfo = $order::find('id = 1');
var_dump($orderInfo);

```

// Order Model code

```
public function setTableName(string $table)
{
    return $this->setSource($table);
}

```
The query structure shows only order_5 data，How can we initialize different tables in a loop and get the data in the specified table。

**Comment by quasipickle** [comment]
_FYI, You can use three backticks to format code blocks._

I believe `setSource()` should be thought of in a static sense.  It applies to the _class_ not the _object_.  So once you set the source for your first `Order`, that locks in the table for all objects of type `Order`.

Using a single model for multiple tables goes against MVC & usage of ORM - where each model is supposed to represent a separate table.  The fact that you have (at least) 6 tables holding identical information that can be accessed using the same object, seems to me like you should just combine all the tables into one `order` table.  I'd bet that would simplify not only this problem, but a lot of other problems related to data access & querying.

Failing that, I think you're going to need to make a separate model class for each table.  You could have a base abstract `Order` class, then extend that class with `Order1`, `Order2`, etc.  In each `Order*` class, set the source in that class' `initialize()` method.
